Password Updating

juliejonak opened the pull request:
Updated the User routes that get and put single user information so they are protected, so that the req.user information is automatically sent with each request. Previously, they were public and not accessing that necessary info, so returning empty responses.

Also updated seed data to provide more information for billing testing, and setup a more thorough logic handling on the input fields for the password update modal.

juliejonak committed:
Settings changes

juliejonak committed:
Merge conflicts sorted

juliejonak committed:
Pulled from development

juliejonak committed:
Added hosted apple domain association file for integrating Apple Pay

juliejonak committed:
Connected router file to applePayRouter

juliejonak committed:
Stripe integration updates:
- API.md: adds Stripe specific routes
- stripe.controllers.js: Creates a subscribe handler
- stripe.router.js: Adjusts the router to use subscribe

juliejonak committed:
Update subscription handling

juliejonak committed:
Updating create subscription method

juliejonak committed:
Reformatted and improved subscribe function

juliejonak committed:
Pulled from development after PR merges

juliejonak committed:
Updated User schema to reflect Stripe fields needed for owners

juliejonak committed:
Billing Plans seeded

juliejonak committed:
UG: User Schema can indicate a required field but Faker data is created even if it does not match type or has a missing required field.

juliejonak committed:
Avoiding merge conflicts in the seed file

juliejonak committed:
Updated password modal to mostly work

juliejonak committed:
Setting up redux store to hold user info for settings

LBWright commented on client/src/Components/Settings/accountForm.jsx:
```diff
@@ -15,6 +16,7 @@ export default class AccountForm extends Component {
 
     componentDidMount = () => {
+        axios.get()
         this.setState({
```
I'm a little late to this party, but no reason to set props onto state. Just gets confusing as components update and props change

LBWright commented on client/src/Components/Settings/actionCreator.js:
```diff
@@ -9,19 +9,37 @@ import config from "../../config/index";
+  return async dispatch => {
+    dispatch({ type: actions.USER_STARTED });
+    try{
+      const getUser = await axios.get(`${config.apiUrl}/api/users/me`)
```
`getUser` might be confusing because the getUser variable is already declared in the upper scope and is the name of the function. Maybe you could just call it `user`?

LBWright commented on client/src/Components/Settings/passwordModal.jsx:
```diff
@@ -1,33 +1,76 @@
     show = () => this.setState({ open: true })
 
     handleChange = (e) => {
+        const { newPassword, oldPassword } = this.state;
```
Could you help me understand this logic?

juliejonak replied:
This handle change is checking the input field that the user is typing in first and will handle logic depending on if they're typing in the new password for the first time or the second one that checks that it matches. If the first one, it sets to state right away; but if the second, it checks that they match before setting to state. Disabled and message are just adjusting what renders for the user's feedback and only allows the Update button to become active when all fields are properly filled.

LBWright commented on server/src/resources/router.js:
```diff
@@ -28,6 +29,10 @@ export const protectedRouter = app => {
     app.use('/api/employees', employeeRouter);
     app.use('/api/stripe', stripeRouter);
     app.use('/api/reservations', reservationsRouter);
+    app.use(
```
Dang. Really cool.

LBWright reviewed (commented):
Love the cleverness and cleanness of the stripe stuff. Really awesome stuff.

juliejonak committed:
Fixes:
- user routers updated to split login and registration public, user get and put to private
updated seed file to have more user seed data for testing billing

juliejonak committed:
Updated prettier lines

jessharrison83 committed:
Merge branch 'development' into feature/settings
